test/main/modules.js

```diff
@@ -25,12 +25,9 @@ QUnit.module('QUnit.module', function () {
       'child-test: before=PC beforeEach=PC',
       'child-afterEach: before=PC beforeEach=PC tester=2',
       'parent-afterEach: before=PC beforeEach=PC tester=2 afterEach=C',
-      'child-after: before=PC beforeEach=PC tester=2 afterEach=CP',
-      'parent-after: before=PC beforeEach=PC tester=2 afterEach=CP after=C'
+      'child-after: before=PC',
+      'parent-after: before=P'
```

Comment:
the parent after hook can't see that `this.after` was set to `C` in the child's after hook, because the assignment becomes one of the child `module.testEnvironment`'s _own_ properties.

This isn't a problem with `before` because the parent's before hook always runs first, and the child _can_ see it, since it inherits from the parent environment.

I think this update to the test is technically correct, but it may highlight that there's additional things we want to assert